lib/docker-compose/models/compose.rb

```diff
@@ -29,6 +34,16 @@ def get_containers_by(params)
   end
 
   #
+  # Select containers based on its given name
+  # (ignore basename)
+  #
+  def get_containers_by_given_name(given_name)
+    @containers.select { |label, container|
+      container.attributes[:name].match(/\w*_#{given_name}_\d+/)
```

Comment:
You're right!
I've fixed it to filter containers based on its directory name as well.
Thanks for the suggestion!